diff --git a/.gitignore b/.gitignore
index a79b33c..390437c 100644
--- a/.gitignore
+++ b/.gitignore
@@ -1,2 +1,3 @@
 /.remote-xcode-server/
 /__pycache__/
+log.txt
diff --git a/linesrenderable.py b/linesrenderable.py
index e418d12..f8b72d2 100644
--- a/linesrenderable.py
+++ b/linesrenderable.py
@@ -144,7 +144,7 @@ class LinesRenderable(Scrollable):
                         if not seg.style:
                             style = None
                         else:
-                            style: Style = seg.style if isinstance(seg.style, Style) else Style(color=seg.style)
+                            style = seg.style if isinstance(seg.style, Style) else Style(color=seg.style)
                         new_seg = Segment(text, style)
                         row.append(new_seg)
                         # if isinstance(seg.style, Style) | isinstance(seg.style, str):
diff --git a/websearch.py b/websearch.py
index 71dc3e5..745d3cb 100644
--- a/websearch.py
+++ b/websearch.py
@@ -152,10 +152,41 @@ x = 5
 curr_idx = 0
 screen_idx = 0
 running = True
+api_key = os.environ.get('BRAVE_API_KEY')
+USER_AGENT = 'CLWebSearchAgent/0.1 (sjberg14@gmail.com) Python-requests/2.33.0'
+search_url = 'https://api.search.brave.com/res/v1/web/search?q='
+query = sys.argv[-1]
+
+
+
+headers = {
+    "Accept": "application/json",
+    "Accept-Encoding": "gzip",
+    "User-Agent": USER_AGENT,
+    "X-Subscription-Token": api_key
+}
 
 if __name__ == '__main__':
-    with open('./results.json', 'r') as f:
-        results_dict = json.load(f)
+
+    offline_mode = False
+
+    #load results dict.  If no argument, load from results.json, otherwise
+    if len(sys.argv) >= 2:
+        #merge all arguments into a single query string, so user doesnt have to use quotes for searches with spaces
+        #i definitely might change how this works later
+        query = ' '.join(sys.argv[1:])
+        print(query)
+        params = {
+            "q": query
+        }
+        resp = requests.get(search_url, params=params, headers=headers)
+        results_dict = resp.json()
+    else:
+        offline_mode = True
+        with open('./results.json', 'r') as f:
+            results_dict = json.load(f)
+
+
     results_list : list[dict] = results_dict['web']['results']
     rows = [[Text('Title'), Text('URL'), Text('Description')]]
     for _ in range(10):
